src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -0,0 +1,607 @@
+            auto ric = ric_attr::get(input).propagate();
+
+            auto order_node = std::dynamic_pointer_cast<opset8::Constant>(pattern_map.at(order_p).get_node_shared_ptr());
+            auto order = order_node->cast_vector<int64_t>();
+
+            int64_t new_axis = std::find(order.begin(), order.end(), ric.get_axis()) - order.begin();
+            ric.set_axis(new_axis);
+
+            ric_attr::set(m.get_match_value(), ric);
+            return true;
+        };
+
+        auto m = std::make_shared<pattern::Matcher>(pattern_root, matcher_name);
+        register_matcher(m, callback);
+    }
+};
+
+class Unsupported : public ngraph::pass::MatcherPass {
+public:
+    Unsupported() {
+        MATCHER_SCOPE(Unsupported);
+        auto pattern_root = pattern::any_input();
+        auto callback = [=](pattern::Matcher& m) {
+            for (const auto & input : m.get_match_root()->input_values()) {
+                if (ric_attr::has(input)) {
+                    auto ric = ric_attr::get(input);
+                    ric.set_can_be_fused(false);
+                    NGRAPH_DEBUG << "Node is unsupported by RIC Fusion: " << *m.get_match_root() << std::endl;
+                }
+            }
+            return true;
+        };
+
+        auto m = std::make_shared<pattern::Matcher>(pattern_root, matcher_name);
+        register_matcher(m, callback);
+    }
+};
+}// namespace prop
+
+namespace fuse {
+namespace {
+bool need_to_erase_ric(const Output<Node> & output) {
+    if (!ric_attr::has(output)) return false;
+    const auto & ric = ric_attr::get(output);
+    return ric.can_be_fused() && ric.is_initial();
+}
+}// namespace
+
+class InsertReverseInputChannel : public ngraph::pass::MatcherPass {
+public:
+    InsertReverseInputChannel() {
+        MATCHER_SCOPE(InsertReverseInputChannel);
+        auto pattern_root = pattern::any_input();
+        auto callback = [](pattern::Matcher& m) {
+            const auto & node = m.get_match_root();
+            for (const auto & input : node->inputs()) {
+                if (!ric_attr::has(input)) continue;
+                const auto & ric = ric_attr::get(input);
+                if (ric.can_be_fused() && ric.is_final()) {
+                    ric(input);
+                }
+            }
+            return false;
+        };
+
+        auto m = std::make_shared<pattern::Matcher>(pattern_root, matcher_name);
+        register_matcher(m, callback);
+    }
+};
+
+class EraseSplitConcat : public ngraph::pass::MatcherPass {
+public:
+    EraseSplitConcat() {
+        MATCHER_SCOPE(EraseSplitConcat);
+        auto input_p = pattern::any_input();
+        auto split_p = pattern::wrap_type<opset8::Split>({input_p, pattern::any_input()});
+        auto pattern_root = pattern::wrap_type<opset8::Concat>({split_p, split_p, split_p}, need_to_erase_ric);
+
+        auto callback = [=](pattern::Matcher& m) {
+            const auto & pattern_map = m.get_pattern_value_map();
```

Comment:
it is shown as not tested for some reason....

Reply:
done.